src/ctree-element-screen/ctree-details-page.html

```diff
@@ -230,6 +231,43 @@
         this._commentSubmitted = true;
       }
 
+      _positiveClicked() {
+        var oldRating = this.userRating;
+          if (this.userRating > 0) {
+            this.userRating = 0;
+          } else {
+            this.userRating = 1;
+          }
+        this.__updateRating(oldRating);
+      }
+
+      _negativeClicked() {
+        var oldRating = this.userRating;
+          if (this.userRating < 0) {
+            this.userRating = 0;
+          } else {
+            this.userRating = -1;
+          }
+        this.__updateRating(oldRating);
+      }
+
+      _isPositive(userRating) {
+        return userRating > 0;
+      }
+
+      _isNegative(userRating) {
+        return userRating < 0;
+      }
+
+      __updateRating(oldRating) {
+        var oldTotalRating = this.rating;
+          if (this.rating != null) {
+            this.rating = this.rating + this.userRating - oldRating;
+          } else {
+            this.rating = this.userRating;
+          }
+      }
```

Comment:
Looks like this is just duplicating code from feedback bar.  Is this really needed?